File: week6/sentimental-readability/readability.py
# TODO
from cs50 import get_string

text = get_string("Text: ")

letters = 0
words = 1
sentences = 0

# Iterating through the text and count the letters, words and sentences.
for c in text:
    if c.isalpha():
        letters += 1

    elif c.isspace():
        words += 1

    elif c == '.' or c == '?' or c == '!':
        sentences += 1

# Calculating the average number of letters and sentences
# We define L & S as float to be precise, but then, we'll round up

L = float(letters) / float(words) * 100
S = float(sentences) / float(words) * 100

# Index formula - plus, we add the round function to round up the result:
index = round(0.0588 * L - 0.296 * S - 15.8)

print(index)

if index >= 16:
    print("Grade 16+")

elif index < 1:
    print("Before Grade 1")

else:
    print("Grade", index)

File: week2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>


int main(void)
{
    string text;
    text  = get_string("Text :");

    int letters = 0;
    //here we start from one as the words will be counted from the first space, which start after the first word. S we need to already cound that first word.
    int words = 1;
    int sentence = 0;


    //Calculating the nb of letters
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }

        //Calculating the nb of words
        else if (isspace(text[i]))
        {
            words++;
        }

        //Calculating the nb of sentences
        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentence++;
        }
    }

    //calculating the average number of letters and sentences
    //we define L & S as float to be precise, but then, we'll round up
    float L = (float) letters / (float) words * 100;
    float S = (float) sentence / (float) words * 100;


    //Index formula - plus, we add the round function to round up the result:
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }

    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    else
    {
        printf("Grade %i\n", index);
    }

}
